app/services/weekend_seconds_interval.rb

```diff
@@ -0,0 +1,24 @@
+class WeekendSecondsInterval < BaseService
+  SECONDS_IN_A_DAY = 864_00
+
+  def initialize(start_date:, end_date:)
+    @start_date = start_date
+    @end_date = end_date
+  end
+
+  def call
+    weekends_as_seconds = weekends_count * SECONDS_IN_A_DAY
```

Comment:
I think this service doesn't support partial dates. e.g. if I call it like:
```
irb(main):011:0> a
=> Sun, 09 Aug 2020 19:18:43 UTC +00:00
irb(main):012:0> b = 5.hours.ago(a)
=> Sun, 09 Aug 2020 14:18:43 UTC +00:00
irb(main):013:0> WeekendSecondsInterval.call(start_date: a, end_date: Time.current)
=> 86400
irb(main):014:0> WeekendSecondsInterval.call(start_date: b, end_date: Time.current)
=> 86400
```

If I'd have had to count the amount of seconds between two dates without weekends, I'd have divided it in 3 parts:
1. seconds from start time til end of first day (unless it's a weekend)
2. seconds from beginning of last day til end time (unless it's a weekend)
3. seconds of full days between start time and end time without weekends

you can get the intermediate days like this `(start_date.tomorrow.to_date..end_date.yesterday.to_date).select { |day| day.on_weekday? }.size * SECONDS_IN_A_DAY`

and then `start_date.on_weekday? start_date.end_of_day - start_date : 0` and `end_date.on_weekday? end_date - end_date.beginning_of_day : 0`

and the sum of the 3 parts is the result.

but this solution might be fine for now

Reply:
i will refactor it then 👍